### PR TITLE
Fix typo

### PR DESCRIPTION
**Pull request by tbroadley**
PR created using https://github.com/tbroadley/github-spellcheck-cli.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1754,8 +1754,8 @@ Other Style Guides
     }
     ```
 
-  <a name="hoisting--named-expresions"></a><a name="14.3"></a>
-  - [14.3](#hoisting--named-expresions) Named function expressions hoist the variable name, not the function name or the function body.
+  <a name="hoisting--named-expressions"></a><a name="14.3"></a>
```
Comment (ljharb) [inline README.md]: while your spelling is indeed correct, links may never change, so this will have to be misspelled forever.

Comment (tbroadley) [reply README.md]: That makes sense. Maybe there's some way to change the link while preserving backwards compatibility with existing links. How about adding an empty anchor with the correct spelling, then fixing the spelling here? Something like:

```md
<a href="hoisting--named-expresions"></a>
- [14.3](#hoisting--named-expressions)
```

I've implemented that change here and it seems to work well: https://github.com/tbroadley/javascript/blob/fix-typo/README.md

Comment (ljharb) [reply README.md]: Sure, we could definitely add an empty anchor for the existing spelling and then correct it - great suggestion! If you restore your PR branch, I'll reopen this PR :-)

Comment (tbroadley) [reply README.md]: Done!

**Comment by ljharb** [comment]
Thank you!

**Comment by tbroadley** [comment]
You're welcome! Thank you for pointing out the issue with my original changeset.

**Pull request merged**